packages/fetchai/skills/weather_client/handlers.py

```diff
@@ -63,14 +65,9 @@ def handle(self, message: Message) -> None:
 
         # recover dialogue
         dialogues = cast(Dialogues, self.context.dialogues)
-        if dialogues.is_belonging_to_registered_dialogue(
-            fipa_msg, self.context.agent_address
-        ):
-            dialogue = cast(
-                Dialogue, dialogues.get_dialogue(fipa_msg, self.context.agent_address)
-            )
-            dialogue.incoming_extend(fipa_msg)
-        else:
+        fipa_dialogue = dialogues.update(fipa_msg)
+        dialogue = cast(Dialogue, fipa_dialogue)
```

Comment:
very confusing with the naming, how about we just do: `dialogue = cast(Dialogue, dialogues.update(fipa_msg))`

Reply:
Done. line 69 was unnecessary